plugins/defaultplugins/aclplugin/vppdump/dump_vppcalls.go

```diff
@@ -26,94 +26,252 @@ import (
+	for identifier, rules := range ruleData {
+		ACLs = append(ACLs, &ACLEntry{
+			Identifier: identifier,
+			ACLDetails: &acl.AccessLists_Acl{
+				Rules:      rules,
+				Interfaces: interfaceData[identifier.ACLIndex],
+			},
+		})
+	}
+
+	return ACLs, nil
+}
+
+// DumpACLRules returns all ruleData for every ACL
+func DumpACLRules(log logging.Logger, vppChannel *govppapi.Channel,
+	timeLog measure.StopWatchEntry) (map[*ACLIdentifier][]*acl.AccessLists_Acl_Rule, error) {
+	// rule map will be returned
+	rules := make(map[*ACLIdentifier][]*acl.AccessLists_Acl_Rule)
+
+	// get all ACLs with IP ruleData
+	IPRuleACLs, err := DumpIPAcls(log, vppChannel, timeLog)
+	if err != nil {
+		return nil, err
+	}
+	// get all ACLs with MACIP ruleData
+	MACIPRuleACLs, err := DumpMacIPAcls(log, vppChannel, timeLog)
+	if err != nil {
+		return nil, err
+	}
+
+	// resolve IP rules for every ACL
+	// Note: currently ACL may have only IP ruleData or only MAC IP ruleData
+	var wasErr error
+	for identifier, IPRules := range IPRuleACLs {
+		var rulesDetails []*acl.AccessLists_Acl_Rule
+
+		if len(IPRules) > 0 {
+			for _, IPRule := range IPRules {
+				ruleDetails, err := getIPRuleDetails(IPRule)
+				if err != nil {
+					log.Error(err)
+					wasErr = err
+				}
+				rulesDetails = append(rulesDetails, ruleDetails)
+			}
+		}
+		rules[identifier] = rulesDetails
+	}
+
+	// resolve MACIP rules for every ACL
+	for identifier, MACIPRules := range MACIPRuleACLs {
+		var rulesDetails []*acl.AccessLists_Acl_Rule
+
+		if len(MACIPRules) > 0 {
+			for _, MACIPRule := range MACIPRules {
+				ruleDetails, err := getMACIPRuleDetails(MACIPRule)
+				if err != nil {
+					log.Error(err)
+					wasErr = err
+				}
+				rulesDetails = append(rulesDetails, ruleDetails)
+			}
+		}
+		rules[identifier] = rulesDetails
+	}
+
+	return rules, wasErr
+}
+
+// DumpACLInterfaces returns a map of ACL indices with interfaces
+func DumpACLInterfaces(indices []uint32, swIfIndices ifaceidx.SwIfIndex, log logging.Logger, vppChannel *govppapi.Channel,
+	timeLog measure.StopWatchEntry) (map[uint32]*acl.AccessLists_Acl_Interfaces, error) {
+	// ACLInterfaceListDump time measurement
 	start := time.Now()
 	defer func() {
 		if timeLog != nil {
 			timeLog.LogTimeEntry(time.Since(start))
 		}
 	}()
 
-	res, err := vppcalls.DumpInterface(swIndex, vppChannel, timeLog)
-	log.Infof("Res: %+v\n", res)
-	if err != nil {
-		return alAcls, err
+	// list of ACL-to-interfaces
+	aclsWithInterfaces := make(map[uint32]*acl.AccessLists_Acl_Interfaces)
+	if swIfIndices == nil {
+		return aclsWithInterfaces, nil
 	}
 
-	if res.SwIfIndex != swIndex {
-		return alAcls, fmt.Errorf(fmt.Sprintf("Returned interface index %d does not match request",
-			res.SwIfIndex))
-	}
+	var interfaceData []*ACLToInterface
 
-	for aidx := range res.Acls {
-		ipACL, err := getIPACLDetails(vppChannel, uint32(aidx))
+	msg := &acl_api.ACLInterfaceListDump{}
+	msg.SwIfIndex = 0xffffffff // dump all
```

Comment:
It's better to initialize the `msg` variable in single call using struct literals, like this: